File: sticky-index/src/main/java/br/com/stickyindex/model/Publisher.java
package br.com.stickyindex.model;

import android.support.v7.widget.RecyclerView;

/**
 * Created by edgar on 5/31/15.
 */
public interface Publisher {
    public void register(Subscriber newObserver);
    public void unregister(Subscriber existentObserver);
    public void notifySubscribers(RecyclerView rv, int dx, int dy);
}

File: sticky-index/src/main/java/br/com/stickyindex/model/Publisher.kt
package br.com.stickyindex.model

import android.support.v7.widget.RecyclerView

/**
 * Created by edgar on 5/31/15.
 */
interface Publisher {
    fun register(newObserver: Subscriber)
    fun unregister(existentObserver: Subscriber)
    fun notifySubscribers(rv: RecyclerView, dx: Int, dy: Int)
}
